### PR TITLE
Add autoindent setting

### PR DESCRIPTION
**Pull request by octref**
Thought it didn't make it to 0.1.8 and was anxious :p

@johnfn 

Yay! We love PRs! 🎊

Please include a description of your change & check your PR against this list, thanks:

- [x] Commit message has a short title & issue references
- [x] Each commit does a logical chunk of work.  
- [x] It builds and tests pass (e.g `gulp tslint`)

More info can be found by clicking the "guidelines for contributing" link above.

### CODE REVIEW

**Comment by rebornix** [comment]
It's Okay that we miss the option in `package.json` as Code just wraps users configuration to an object (json to object) so everything works well except autocomplete for this option :)

File: package.json [hunk 1]
```diff
@@ -206,7 +206,13 @@
                     "type": "boolean",
                     "description": "When there is a previous search pattern, highlight all its matches.",
                     "default": false
+                },
+                "vim.autoindent": {
+                    "type": "boolean",
+                    "description": "Keep indentation when doing cc or S in normal mode to replace a line.",
```
Comment (rebornix) [inline package.json]: we'd better use Vim's official description, see http://vimdoc.sourceforge.net/htmldoc/options.html#'autoindent'

**Comment by octref** [comment]
@rebornix 
Addressed. Take a look.

**Comment by rebornix** [comment]
@octref thanks !

**Pull request merged**